File: ytrue-application/ytrue-application-admin/src/main/java/com/ytrue/modules/system/model/po/SysUser.java
package com.ytrue.modules.system.model.po;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.ytrue.common.constant.TimeFormat;
import com.ytrue.common.constant.TimeZone;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import org.springframework.format.annotation.DateTimeFormat;

import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import java.time.LocalDateTime;

/**
 * @author ytrue
 * @description: SysUser
 * @date 2022/12/7 15:20
 */
@Data
@TableName("sys_user")
public class SysUser {

    @TableId
    @Schema(title = "id")
    private Long id;

    @Schema(title = "用户部门Id")
    private Long deptId;

    @NotBlank
    @Schema(title = "用户名称")
    private String username;

    @NotBlank
    @Schema(title = "用户昵称")
    private String nickName;

    @Email
    @NotBlank
    @Schema(title = "邮箱")
    private String email;

    @NotBlank
    @Schema(title = "电话号码")
    private String phone;

    @Schema(title = "用户性别")
    private Integer gender;

    @Schema(title = "头像真实名称")
    private String avatarName;

    @Schema(title = "头像存储的路径")
    private String avatarPath;

    @Schema(title = "密码")
    @JsonIgnore
    private String password;

    @NotNull
    @Schema(title = "是否启用")
    private Boolean status;

    @Schema(title = "是否为admin账号")
    private Boolean isAdmin;

    @Schema(title = "最后修改密码的时间")
    private LocalDateTime pwdResetTime;

    @TableField(fill = FieldFill.INSERT)
    @Schema(title = "创建人", hidden = true)
    private String createBy;

    @TableField(fill = FieldFill.UPDATE)
    @Schema(title = "更新人", hidden = true)
    private String updateBy;

    @TableField(fill = FieldFill.INSERT)
    @Schema(title = "创建时间")
    @JsonFormat(timezone = TimeZone.GMT8,pattern = TimeFormat.DATE_TIME_FORMAT)
    @DateTimeFormat(pattern = TimeFormat.DATE_TIME_FORMAT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.UPDATE)
    @Schema(title = "更新时间")
    @JsonFormat(timezone = TimeZone.GMT8,pattern = TimeFormat.DATE_TIME_FORMAT)
    @DateTimeFormat(pattern = TimeFormat.DATE_TIME_FORMAT)
    private LocalDateTime updateTime;
}

File: ytrue-admin-server-csharp/Ytrue.Models/Entities/System/SysUser.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Ytrue.Models.Entities.System;

/// <summary>
/// 系统用户实体
/// </summary>
[Table("sys_user")]
public class SysUser
{
    /// <summary>
    /// 用户ID
    /// </summary>
    [Key]
    public long Id { get; set; }

    /// <summary>
    /// 部门ID
    /// </summary>
    [Column("dept_id")]
    public long? DeptId { get; set; }

    /// <summary>
    /// 用户名
    /// </summary>
    [Column("username")]
    public string? UserName { get; set; }

    /// <summary>
    /// 昵称
    /// </summary>
    [Column("nick_name")]
    public string? NickName { get; set; }

    /// <summary>
    /// 性别: 0=女, 1=男
    /// </summary>
    [Column("gender")]
    public bool Gender { get; set; }

    /// <summary>
    /// 手机号码
    /// </summary>
    [Column("phone")]
    public string? Phone { get; set; }

    /// <summary>
    /// 邮箱
    /// </summary>
    [Column("email")]
    public string? Email { get; set; }

    /// <summary>
    /// 头像地址
    /// </summary>
    [Column("avatar_name")]
    public string? AvatarName { get; set; }

    /// <summary>
    /// 头像真实路径
    /// </summary>
    [Column("avatar_path")]
    public string? AvatarPath { get; set; }

    /// <summary>
    /// 密码
    /// </summary>
    [Column("password")]
    public string? Password { get; set; }

    /// <summary>
    /// 是否为admin账号: 1=是, 0=否
    /// </summary>
    [Column("admin")]
    public bool? Admin { get; set; }

    /// <summary>
    /// 状态: 1=启用, 0=禁用
    /// </summary>
    [Column("status")]
    public bool? Status { get; set; }

    /// <summary>
    /// 创建者
    /// </summary>
    [Column("create_by")]
    public string? CreateBy { get; set; }

    /// <summary>
    /// 更新者
    /// </summary>
    [Column("update_by")]
    public string? UpdateBy { get; set; }

    /// <summary>
    /// 修改密码的时间
    /// </summary>
    [Column("pwd_reset_time")]
    public DateTime? PwdResetTime { get; set; }

    /// <summary>
    /// 创建日期
    /// </summary>
    [Column("create_time")]
    public DateTime? CreateTime { get; set; }

    /// <summary>
    /// 更新时间
    /// </summary>
    [Column("update_time")]
    public DateTime? UpdateTime { get; set; }
}
